src/main/java/com/auth0/jwk/JwkProviderBuilder.java

```diff
@@ -82,17 +81,28 @@ public JwkProviderBuilder cached(boolean cached) {
      *
      * @param cacheSize number of jwk to cache
      * @param expiresIn amount of time the jwk will be cached
-     * @param unit      unit of time for the expire of jwk
      * @return the builder
      */
-    public JwkProviderBuilder cached(long cacheSize, long expiresIn, TimeUnit unit) {
+    public JwkProviderBuilder cached(long cacheSize, Duration expiresIn) {
         this.cached = true;
         this.cacheSize = cacheSize;
         this.expiresIn = expiresIn;
-        this.expiresUnit = unit;
         return this;
     }
 
+    /**
+     * Enable the cache specifying size and expire time.
+     *
+     * @param cacheSize number of jwk to cache
+     * @param expiresIn amount of time the jwk will be cached
+     * @param unit      unit of time for the expire of jwk
+     * @return the builder
+     * @deprecated prefer the use of {@link JwkProviderBuilder#cached(long, Duration)}
```

Comment:
Remove this line. There's no reason to deprecate this method today. If anything, reference to it via a `@see` annotation.